### ISSUE TITLE
output config: Ability to assign all exports to arbitrary object, creating the object if it doesn't exist, re-using it otherwise.

### ISSUE THREAD

**Issue by trusktr** [webpack/webpack#11571]
## Feature request

**What is the expected behavior?**

Some sort of `library`/`libraryTarget` option to allow us to assign to an object if it exists, but create it otherwise.

For example, with `libraryTarget: 'assign'` the output is like this:

```js
TheObject = _entry_return_;
```

but instead, we could make it do this:

```js
Object.assign(TheObject || (TheObject = {}), _entry_return_)
```

**What is motivation or use case for adding/changing the behavior?**

The only way to get _close_ to this is using an array for `library`, but it isn't exactly what is desired. For example, if we have `library: ['TheObject', 'SubObject']`, then _only while having at least two items in the array_ the first object will be created. 

Basically, [this question](https://stackoverflow.com/questions/54830256/webpack-use-object-assign-in-output-librarytarget) on StackOverflow also describes the missing feature that is desired.

**How should this be implemented in your opinion?**

Similar to above snippet.

Maybe the new option is `libraryTarget: 'assign2'` (so it doesn't change behavior of the current `assign`)?

**Are you willing to work on this yourself?**
Maybe

**Comment by sokra** [comment]
Yep send a PR for a new library type. Maybe with a better name, e.g. `assign-copy` or `assign-properties`

**Comment by sag1v** [comment]
Hi, is this something that will get released on all versions or only latest version?
Is there a workaround to get this result in v3 and v4, like a plugin or a hook that can alter the behavior of storing the properties on the `libraryTarget` variable?

Adding [my SO question](https://stackoverflow.com/questions/65069303/webpack-librarytarget-merge-properties-to-an-existing-variable?noredirect=1#comment115041001_65069303) for reference

**Comment by sag1v** [comment]
@sokra What versions was it implemented in?

**Comment by alexander-akait** [comment]
@sag1v Only webpack@5, sorry, what is the problem to update?

**Comment by sag1v** [comment]
@alexander-akait Our micro-frontends architecture is dependent on this and we can't update it for all services, at least not in one go.

**Comment by alexander-akait** [comment]
@sag1v in this case you can fork webpack@4 and put changes in fork
